src/averageAges.js

```diff
@@ -37,7 +38,12 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f'
+  && (!withChildren || people.some(p => p.mother === person.name)));
+  const totalAge = women.reduce((total, person) =>
+    total + (person.died - person.born), 0);
+
+  return totalAge / women.length;
```

Comment:
really difficult to read, please improve formatting accordingly to the checklist